pykafka/rdkafka/_rd_kafkamodule.c

```diff
@@ -0,0 +1,448 @@
+    } else {
+        Py_DECREF(stop_result);
+    }
+    self->ob_type->tp_free(self);
+}
+
+
+static const char Consumer_configure__doc__[] =
+"Set up and populate the rd_kafka_(topic_)conf_t\n"
+"\n"
+"Somewhat inelegantly (for the benefit of code reuse, whilst avoiding some\n"
+"harrowing partial binding for C functions) this requires that you call it\n"
+"twice, once with a `conf` list only, and again with `topic_conf` only.\n"
+"\n"
+"Repeated calls work incrementally; you can wipe configuration completely\n"
+"by calling Consumer_stop()\n";
+static PyObject *
+Consumer_configure(Consumer *self, PyObject *args, PyObject *kwds) {
+    char *keywords[] = {"conf", "topic_conf", NULL};
+    PyObject *conf = NULL;
+    PyObject *topic_conf = NULL;
+    if (! PyArg_ParseTupleAndKeywords(args,
+                                      kwds,
+                                      "|OO",
+                                      keywords,
+                                      &conf,
+                                      &topic_conf)) return NULL;
+
+    if ((conf && topic_conf) || (!conf && !topic_conf)) {
+        set_PyRdKafkaError(
+            RD_KAFKA_RESP_ERR__FAIL,
+            "You need to specify *either* `conf` *or* `topic_conf`.");
+        return NULL;
+    }
+    if (self->rdk_handle) {
+        set_PyRdKafkaError(
+            RD_KAFKA_RESP_ERR__FAIL,
+            "Cannot configure: seems instance was started already?");
+        return NULL;
+    }
+
+    if (! self->rdk_conf) self->rdk_conf = rd_kafka_conf_new();
+    if (! self->rdk_topic_conf) {
+        self->rdk_topic_conf = rd_kafka_topic_conf_new();
+    }
+
+    PyObject *conf_or_topic_conf = topic_conf ? topic_conf : conf;
+    Py_ssize_t i, len = PyList_Size(conf_or_topic_conf);
+    for (i = 0; i != len; ++i) {
+        PyObject *conf_pair = PyList_GetItem(conf_or_topic_conf, i);
+        const char *name = NULL;
+        const char *value =  NULL;
+        if (! PyArg_ParseTuple(conf_pair, "ss", &name, &value)) return NULL;
+
+        char errstr[512];
+        rd_kafka_conf_res_t res;
+        if (topic_conf) {
+            res = rd_kafka_topic_conf_set(
+                    self->rdk_topic_conf, name, value, errstr, sizeof(errstr));
+        } else {
+            res = rd_kafka_conf_set(
+                    self->rdk_conf, name, value, errstr, sizeof(errstr));
+        }
+        if (res != RD_KAFKA_CONF_OK) {
+            set_PyRdKafkaError(RD_KAFKA_RESP_ERR__FAIL, errstr);
+            return NULL;
+        }
+    }
+    Py_INCREF(Py_None);
+    return Py_None;
+}
+
+
+static PyObject *
+Consumer_start(Consumer *self, PyObject *args, PyObject *kwds) {
+    char *keywords[] = {
+        "brokers",
+        "topic_name",
+        "partition_ids",
+        "start_offsets",  // same order as partition_ids
+        NULL};
+    const char *brokers = NULL;
+    const char *topic_name = NULL;
+    PyObject *partition_ids = NULL;
+    PyObject *start_offsets= NULL;
```

Comment:
Most of that PEP7 stuff looks fine (although the requirement to write valid C89 is going to be tricky - gcc won't complain, and I won't always spot C99-isms).

The one thing I do have a bit of a beef with is also C89-related: I think putting all variable declarations at the top of the block really decreases code readability. So.... can I violate that one please :)